src/registrar/models/contact.py

```diff
@@ -59,6 +59,16 @@ def get_formatted_name(self):
         names = [n for n in [self.first_name, self.middle_name, self.last_name] if n]
         return " ".join(names) if names else "Unknown"
 
+    def save(self, *args, **kwargs):
+        # Call the parent class's save method to perform the actual save
+        super().save(*args, **kwargs)
+
+        # Update the related User object's first_name and last_name
+        if self.user:
```

Comment:
**question (non-blocking):**: This is very clever to keep user objects in sync with contacts rather than vice versa. But for an existing user who does log in with identity verification, if their `given_name` and `family_name` have changed with Login.gov (can this ever happen?) will their `User` object get updated by not their `Contact` object? Is that what we want?

Reply:
No, this will not get updated from login.gov, per my other comment.  It is not that we can't handle receiving this from login.gov, it is just that login.gov is not sending it when user logs in with IAL1.